doc/src/modules/vector/basics.rst

```diff
@@ -0,0 +1,254 @@
+  >>> v - N.j
+  2*N.i
+  >>> type(v - N.j)
+  <class 'sympy.vector.vector.VectorMul'>
+
+What about a zero vector? It can be accessed using the ``zero``
+attribute assigned to class ``Vector``. Since the notion of a zero
+vector remains the same regardless of the coordinate system in 
+consideration, we use ``Vector.zero`` wherever such a quantity is
+required.
+
+  >>> from sympy.vector import Vector
+  >>> Vector.zero
+  0
+  >>> type(Vector.zero)
+  <class 'sympy.vector.vector.VectorZero'>
+  >>> N.i + Vector.zero
+  N.i
+  >>> Vector.zero == 2*Vector.zero
+  True
+
+All the classes shown above - ``BaseVector``, ``VectorMul``, 
+``VectorAdd`` and ``VectorZero`` are subclasses of ``Vector``.
+
+You should never have to instantiate objects of any of the
+subclasses of ``Vector``. Using the ``BaseVector`` instances assigned to a
+``CoordSysCartesian`` instance and (if needed) ``Vector.zero``
+as building blocks, any sort of vectorial expression can be constructed
+with the basic mathematical operators ``+``, ``-``, ``*``.
+and ``/``.
+
+  >>> v = N.i - 2*N.j
+  >>> v/3
+  1/3*N.i + (-2/3)*N.j
+  >>> v + N.k
+  N.i + (-2)*N.j + N.k
+  >>> Vector.zero/2
+  0
+  >>> (v/3)*4
+  4/3*N.i + (-8/3)*N.j
+
+
+In addition to the elementary mathematical operations, the vector 
+operations of ``dot`` and ``cross`` can also be performed on 
+``Vector``.
+
+  >>> v1 = 2*N.i + 3*N.j - N.k
+  >>> v2 = N.i - 4*N.j + N.k
+  >>> v1.dot(v2)
+  -11
+  >>> v1.cross(v2)
+  (-1)*N.i + (-3)*N.j + (-11)*N.k
+  >>> v2.cross(v1)
+  N.i + 3*N.j + 11*N.k
+
+The ``&`` and ``^`` operators have been overloaded for the
+``dot`` and ``cross`` methods respectively.
+
+  >>> v1 & v2
+  -11
+  >>> v1 ^ v2
+  (-1)*N.i + (-3)*N.j + (-11)*N.k
+
+In addition to these operations, it is also possible to compute the
+outer products of ``Vector`` instances in :mod:`sympy.vector`. More
+on that in a little bit.
+
+
+SymPy operations on Vectors
+===========================
+
+The SymPy operations of ``simplify``, ``trigsimp``, ``diff``,
+and ``factor`` work on ``Vector`` objects, with the standard SymPy API.
+
+In essence, the methods work on the measure numbers(The coefficients 
+of the basis vectors) present in the provided vectorial expression.
+
+  >>> from sympy.abc import a, b, c
+  >>> from sympy import sin, cos, trigsimp, diff
+  >>> v = (a*b + a*c + b**2 + b*c)*N.i + N.j
+  >>> v.factor()
+  ((a + b)*(b + c))*N.i + N.j
+  >>> v = (sin(a)**2 + cos(a)**2)*N.i - (2*cos(b)**2 - 1)*N.k
+  >>> trigsimp(v)
+  N.i + (-cos(2*b))*N.k
+  >>> v.simplify()
+  N.i + (-cos(2*b))*N.k
+  >>> diff(v, b)
+  (4*sin(b)*cos(b))*N.k
+  >>> from sympy import Derivative
+  >>> Derivative(v, b).doit()
+  (4*sin(b)*cos(b))*N.k
+
+``Integral`` also works with ``Vector`` instances, similar to
+``Derivative``.
+
+  >>> from sympy import Integral
+  >>> v1 = a*N.i + sin(a)*N.j - N.k
+  >>> Integral(v1, a)
+  (Integral(a, a))*N.i + (Integral(sin(a), a))*N.j + (Integral(-1, a))*N.k
```

Comment:
As I mentioned before, I have tried to make it easy to understand the measure numbers and basis vectors involved in a vector expression by using brackets. This 'auto-evaluation' was put in for two reasons-
1. The way in which a vectorial expression gets integrated by a scalar is satisfied explicitely by this format.
2. It was causing some errors if kept in an Integral instance(like VectorMuls being converted into Muls), since that class does not deal with vectors. Therefore, by making it a vector instance immediately, these problems are avoided too.